src/client/graphql-content-provider.ts

```diff
@@ -159,7 +166,8 @@ export class GraphQLContentProvider implements TextDocumentContentProvider {
 
         this.getEndpointName(endpointNames).then(endpointName => {
           const endpoint = projectConfig!.endpointsExtension!.getEndpoint(
-            endpointName
+            endpointName,
+            this.env
```

Comment:
This is the fix, which consists of passing the env variables imported from `.env` file. Not doing so will result in defaulting to `process.env`, which afaiu is different from the process of the server.
https://github.com/prisma/graphql-config/blob/master/src/extensions/endpoints/EndpointsExtension.ts#L65-L69